_pages/documentation/general_docs/kconfig_build_system.md

```diff
@@ -0,0 +1,198 @@
+---
+layout: documentation
+title: Kconfig Build System
+doc: gem5 documentation
+parent: kconfig_build_system
+permalink: /documentation/general_docs/kconfig_build_system/
+---
+
+For advanced users who need to build gem5 with multiple ISAs or customized the
+build options, such as ruby memory protocol. Users need to familiar with
+Kconfig system.
+
+## Build GEM5 with Kconfig Build System
+
+```bash
+scons [OPTIONS] Kconfig_command TARGET
+```
+
+Supported kconfig commands include
+
+- defconfig
+- setconfig
+- menuconfig
+- guiconfig
+- listnewconfig
+- oldconfig
+- olddefconfig
+- savedefconfig
+
+The most common use of options are `defconfig`, `setconfig` and `menuconfig`.
+
+Use `scons --help` to list these commands, and for additional information.
+
+To build gem5 with kconfig, there are now two steps.
+
+Initial configuration
+
+```bash
+scons defconfig gem5_build build_opts/ALL
+```
+
+This will create a configuration for the gem5_build build directory based on
+`build_opts/ALL`. The configuration is stored within the build directory.
+
+To build a target in a configured build directory, run a scons command similar
+to before.
+
+```bash
+scons gem5_build/gem5.opt
+```
+
+To build gem5 kconfig with customized kconfig options, the additional step
+is required between **initial configuration** and **building the target**
+
+- Directly set kconfig options
+
+```bash
+scons setconfig gem5_build USE_KVM=y
+```
+
+- set kconfig options by `menuconfig`
+
+```bash
+scons menuconfig gem5_build
+```
+
+## Details of Kconfig Commands
+
+### defconfig
+
+Set up a config using values specified in a defconfig file, or if no value is
+given, use the default. The second argument specifies the defconfig file. All
```

Comment:
Where would I find the default value? How is this specified?

Reply:
Added in the doc description